src/detect-file-data.c

```diff
@@ -84,7 +84,7 @@ static int DetectFiledataSetup (DetectEngineCtx *de_ctx, Signature *s, char *str
         return -1;
     }
 
-    if ((s->init_flags & SIG_FLAG_INIT_FLOW) &&
+    if ((s->init_flags & SIG_FLAG_INIT_FLOW) && !DetectProtoContainsProto(&s->proto, IPPROTO_TCP) &&
```

Comment:
I don't see how this fixes anything. It just effectively disables this whole test, as all valid sigs that use 'alert http', 'alert smtp' or 'alert tcp' will fail to meet this new condition.